src/ios-deploy/ios-deploy.m

```diff
@@ -959,13 +963,27 @@ void write_lldb_prep_cmds(AMDeviceRef device, CFURLRef disk_app_url) {
 void
 server_callback (CFSocketRef s, CFSocketCallBackType callbackType, CFDataRef address, const void *data, void *info)
 {
-    if (CFDataGetLength (data) == 0) {
+    char buffer[0x4000];
+    int bytesRead = AMDServiceConnectionReceive(dbgServiceConnection, buffer, sizeof(buffer));
+    if (bytesRead == 0)
+    {
         // close the socket on which we've got end-of-file, the server_socket.
         CFSocketInvalidate(s);
         CFRelease(s);
         return;
     }
-    write(CFSocketGetNative(lldb_socket), CFDataGetBytePtr(data), CFDataGetLength(data));
+    write(CFSocketGetNative (lldb_socket), buffer, bytesRead);
+    if (bytesRead == sizeof(buffer))
```

Comment:
Should this be: `while (bytesRead == sizeof(buffer)) { ... }`?

If we get a packet where size = $BUF_SIZE + 1, this will:
- read $BUF_SIZE bytes
- write $BUF_SIZE bytes to socket
- read 1 byte
- write 1 byte to socket
- wait to read more data which may not be there 

With the singular while loop we can stop reading after we've consumed all the data.

Reply:
Yeah, removed the redundant if in favor of while.

Reply:
Yeah, removed the redundant if in favor of while.